pkg/operation/botanist/addons.go

```diff
@@ -243,7 +244,7 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 			"blackbox-exporter": blackboxExporter,
 		}, b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting),
 		"network-policies":        networkPolicyConfig,
-		"podsecuritypolicies":     common.GenerateAddonConfig(podSecurityPolicies, true),
+		"podsecuritypolicies":     common.GenerateAddonConfig(podSecurityPolicies, !b.Shoot.PSPDisabled),
```

Comment:
This will delete the `gardener.unprivileged` and `gardener.privileged` PSPs, right?

From https://github.com/gardener/gardener/issues/5250:
> Adapt mutating fields handled by PSPs in gardener and extensions components in the PodSpec so that we can drop all the gardener deployed PSPs when the user disables the PodSecurityPolicy admission plugin.

There are Pods that implicitly use the `gardener.unprivileged` PSP. Don't they need to be adapted to now contained the previously mutated fields now in their Pod spec?

Reply:
hmm..Thanks for pointing out, I just checked and saw that when `allowPrivilegedContainers=false`
```
kube-system blackbox-exporter-6f4d6f746b-597c5
kube-system blackbox-exporter-6f4d6f746b-59wmk
kube-system coredns-8455d66f48-5fn94
kube-system coredns-8455d66f48-xx8wh
kube-system metrics-server-685549996d-2mj4x
```
these components use `gardener.unprivileged` and it does mutate some fields.
So given that these components can run fine with these fields set,
can we adapt the PodSpec accordingly, regardless of whether PSP plugin is disabled?

Reply:
I think you can adapt the Pod spec regardless of whether the PodSecurityPolicy admission plugin is enabled/disabled. To my understanding the component will just run "more secure" by default (not only when PodSecurityPolicy admission plugin is enabled and `allowPrivilegedContainers=false`). Do you have a concern?
This is what you did also in https://github.com/gardener/gardener-extension-networking-calico/pull/203, right?

Reply:
Yes, but there it was dedicated PSPs.

Reply:
I think it should not matter if it is dedicated PSPs or the `gardener.unprivileged` one. Right?

Reply:
If the pod is fine to run with this PodSpec, it shouldn't matter.

Reply:
As agreed in a call, let's also adapt the specs of Pods using the `gardener.unprivileged` PSP. For example the metrics-server gets the following additional fields:
```diff
   securityContext:
     fsGroup: 65534
     runAsUser: 65534
+    seccompProfile:
+      type: RuntimeDefault
+    supplementalGroups:
+    - 1
```

Reply:
coredns is already using these specs. Adapted others in gardener@63fee6e

Reply:
> `coredns` is already using these specs.

For coredns I see that the Pod securityContext looks like:

https://github.com/gardener/gardener/blob/4b326d9eaefc4ef378fb2c5ad4d5393b5f03ee98/pkg/operation/botanist/component/coredns/coredns.go#L430-L433

But the mutated Pod securityContext looks like:
```diff
   securityContext:
+    fsGroup: 1
     runAsNonRoot: true
     runAsUser: 65534
+    seccompProfile:
+      type: RuntimeDefault
+    supplementalGroups:
+    - 1
```

Reply:
For the kubernetes-dashboard I see Pod securityContext looks like:
https://github.com/gardener/gardener/blob/3ae21a9907988bb9b73a1210c989f2777d794f8f/charts/shoot-addons/charts/kubernetes-dashboard/templates/deployment-dashboard.yaml#L31-L35

But the mutated Pod securityContext looks like :

```diff
  securityContext:
+    fsGroup: 1
     runAsGroup: 2001
     runAsUser: 1001
+    seccompProfile:
+      type: RuntimeDefault
+    supplementalGroups:
+    - 1
```

Reply:
Similarly dashboard-metric-scaper also needs to be adapted 
https://github.com/gardener/gardener/blob/3ae21a9907988bb9b73a1210c989f2777d794f8f/charts/shoot-addons/charts/kubernetes-dashboard/templates/deployment-metrics-scraper.yaml#L50-L52